source/acquaman/AMTimedScanActionControllerAssembler.cpp

```diff
@@ -0,0 +1,98 @@
+#include "AMTimedScanActionControllerAssembler.h"
+
+#include "actions3/AMListAction3.h"
+#include "actions3/AMLoopAction3.h"
+#include "actions3/AMActionSupport.h"
+#include "actions3/actions/AMAxisStartedAction.h"
+#include "actions3/actions/AMAxisFinishedAction.h"
+#include "actions3/actions/AMAxisValueFinishedAction.h"
+#include "beamline/AMDetectorTriggerSource.h"
+#include "actions3/actions/AMWaitAction.h"
+
+AMTimedScanActionControllerAssembler::AMTimedScanActionControllerAssembler(double acquisitionTime, double timePerAcquisition, int iterations, QObject *parent)
+	: AMScanActionControllerScanAssembler(parent)
+{
+	acquisitionTime_ = acquisitionTime;
+	timePerAcquisition_ = timePerAcquisition;
+	iterations_ = iterations;
+}
+
+AMTimedScanActionControllerAssembler::~AMTimedScanActionControllerAssembler()
+{
+
+}
+
+bool AMTimedScanActionControllerAssembler::generateActionTreeImplmentation()
+{
+	AMLoopAction3 *actionTree = new AMLoopAction3(iterations_);
```

Comment:
If you have a better set of definitions I'll switch them.  They seemed the most appropriate at the time.